araproc/analysis/correlation.py

```diff
@@ -0,0 +1,114 @@
+import ROOT
+import numpy as np
+import math
+import os
+
+# HELP HERE: constants that we might want to store elsewhere, or can I retrieve them somehow?
+channel_num = 16
+
+def get_surface_correlation_ratio(corr_map, station_id, z_thresh):
+    """
+    Calculates the ratio of the maximum surface correlation value within a specific
+    theta range to the overall peak correlation value. The theta range is determined
+    based on the average antenna z-coordinate and a threshold.
+
+    Parameters
+    ----------
+    corr_map : dict
+        A dictionary containing:
+        - 'map': ROOT.TH2D histogram with correlation values mapped by theta and phi.
+        - 'corr': The overall peak correlation value.
+        - 'radius': The correlation radius for the station.
+    station_id : int
+        The station ID for which to calculate the correlation ratio.
+    z_thresh : float
+        The threshold value to define the lower bound of the theta range.
+
+    Returns
+    -------
+    surf_corr_ratio : float
+        The ratio of the maximum surface correlation value within the theta range 
+        to the overall peak correlation.
+    max_surf_theta : float
+        The theta angle corresponding to the maximum surface correlation value.
+    max_surf_phi : float
+        The phi angle corresponding to the maximum surface correlation value.
+    """
+    
+    # Extract the radius from corr_map
+    corr_radius = float(corr_map.get("radius", None))
+    if corr_radius is None:
+        raise ValueError("Radius not found in corr_map.")
+
+    # Initialize AraGeomTool instance to retrieve antenna information
+    geomTool = ROOT.AraGeomTool.Instance()
+
+    # Collect z-coordinates of all antennas to calculate the average z-coordinate
+    z = []
+    for ant in range(channel_num):
+        # Retrieve each antenna's location for the given station
+        ant_location = geomTool.getStationInfo(station_id).getAntennaInfo(ant).antLocation
+        z.append(ant_location[2])  # Append only the z-coordinate
+
+    # Calculate the average z-coordinate of the antennas
+    avg_z = sum(z) / len(z)
+
+    # Check if the radius is large enough to visualize the surface
+    if corr_radius < abs(avg_z):
+        raise RuntimeError("Surface not visible: radius is smaller than average z coordinate of the antennas.")
+
+    # Calculate the angles at the surface and z threshold relative to the horizontal
+    theta_at_surface = math.degrees(math.asin(abs(avg_z) / corr_radius))
```

Comment:
I think this should be `acos`?

Reply:
`theta` in the map is measured in elevation [90, -90], so I think it's fine?

Reply:
ohhh sorry i thought it was the zenith angle. Okay then I agree you have it right.